src/main/java/com/microsoft/aad/msal4j/TokenCache.java

```diff
@@ -300,16 +302,92 @@ static private AppMetadataCacheEntity createAppMetadataCacheEntity(TokenRequestE
                         build())) {
             try {
                 lock.readLock().lock();
+                Set<IAccount> rootAccounts = new HashSet<>();
 
-                return accounts.values().stream().
+                //Filter accounts map into two sets: a set of home accounts (local OID = home OID),
+                // and a set of tenant profiles (local OID != home OID)
+                Set<AccountCacheEntity> homeAccounts = accounts.values().stream().
                         filter(acc -> environmentAliases.contains(acc.environment())).
-                        collect(Collectors.mapping(AccountCacheEntity::toAccount, Collectors.toSet()));
+                        filter(acc -> acc.homeAccountId().contains(acc.localAccountId())).
```

Comment:
homeAccountID is in the format of objectID.tenantID, but the localAccountID only corresponds to the objectID,: for an account the local object ID matches the one in the homeAccountID, and for a tenant profile does not. I could've done something like `localAccountId.equals(homeAccountId.split("\\.")[0])`, but I figured this would be more readable.